src/cpp/shared/combat/army.cpp

```diff
@@ -2456,3 +2478,188 @@ int army::GetStraightLineDistanceToHex(int hex) {
+    int stackNumXOffset;
+    if(this->creature.creature_flags & TWO_HEXER) {
+      if(this->facingRight == 1) {
+        stackNumXOffset = x + 53;
+        hexInFront = this->occupiedHex + 2;
+      } else {
+        stackNumXOffset = x - 73;
+        hexInFront = this->occupiedHex - 2;
+      }
+    } else if(this->facingRight == 1) {
+      stackNumXOffset = x + 9;
+      hexInFront = this->occupiedHex + 1;
+    } else {
+      stackNumXOffset = x - 29;
+      hexInFront = this->occupiedHex - 1;
+    }
+    bool hexInFrontClear = gpCombatManager->combatGrid[hexInFront].unitOwner != -1;
+    int creatureStackNumXOffset = this->frameInfo.stackNumDispXShift[1 - this->facingRight];
+    if(hexInFrontClear && creatureStackNumXOffset > 0)
+      creatureStackNumXOffset = 0;
+    if(this->facingRight == 1)
+      stackNumXOffset += creatureStackNumXOffset;
+    else
+      stackNumXOffset -= creatureStackNumXOffset;
+
+    int offsetY;
+    if(this->facingRight == 1)
+      offsetY = y - 11;
+    else
+      offsetY = y - 23;
+    int numPosEffects = 0;
+    int numNegEffects = 0;
+    for(int i = 0; i < NUM_SPELL_EFFECTS; i++) {
+      if(this->effectStrengths[i]) {
+        switch(i) {
+          case EFFECT_HASTE:
+          case EFFECT_BLESS:
+          case EFFECT_DRAGON_SLAYER:
+          case EFFECT_BLOOD_LUST:
+          case EFFECT_SHIELD:
+          case EFFECT_ANTI_MAGIC:
+          case EFFECT_STONESKIN:
+          case EFFECT_STEELSKIN:
+          case EFFECT_FORCE_SHIELD:
+            ++numPosEffects;
+            break;
+          default:
+            ++numNegEffects;
+            break;
+        }
+      }
+    }
+
+    int inRedrawZone;
+    if(giSpellEffectShowType && isIdle && this->numActiveEffects > 0) {
+      if(giSpellEffectShowType == 1) {
+        inRedrawZone = gpCombatManager->combatScreenIcons[1]->CombatClipDrawToBuffer(stackNumXOffset, offsetY, 11, &this->stackSizeDispBounds, 0, 237, 0, 0);
+      } else {
+        int numImageIdx = 12;
+        if(numPosEffects <= 0 || numNegEffects <= 0) {
+          if(numNegEffects > 0)
+            numImageIdx = 14;
+        } else {
+          numImageIdx = 13;
+        }
+        inRedrawZone = gpCombatManager->combatScreenIcons[1]->CombatClipDrawToBuffer(stackNumXOffset, offsetY, numImageIdx, &this->stackSizeDispBounds, 0, 0, 0, 0);
+      }
+    } else {
+      inRedrawZone = gpCombatManager->combatScreenIcons[1]->CombatClipDrawToBuffer(stackNumXOffset, offsetY, 10, &this->stackSizeDispBounds, 0, 0, 0, 0);
+    }
+    if(gIronfistExtra.combat.stack.forceShieldHP[this] > 0) {
+      const int FORCE_SHIELD_ICON_X_OFFSET = 2;
+      const int FORCE_SHIELD_ICON_Y_OFFSET = -28;
+      const int FORCE_SHIELD_HP_BOX_Y_OFFSET = -24;
+
+      icon* shieldIcon = gpResourceManager->GetIcon("SPELLINF.ICN");
+      shieldIcon->CombatClipDrawToBuffer(stackNumXOffset + FORCE_SHIELD_ICON_X_OFFSET, offsetY + FORCE_SHIELD_ICON_Y_OFFSET, 10, &this->stackSizeDispBounds, 0, 0, gColorTableRed, 0);
```

Comment:
Does this work when the player is the defender?

Reply:
Good catch. But yes it works, just checked